src/runtime/mod.rs

```diff
@@ -105,9 +90,17 @@ impl Runtime {
             }
         };
 
-        tokio::pin!(future);
+        local.spawn_local(drive);
 
-        self.local.spawn_local(drive);
+        Ok(Runtime { local, rt })
+    }
+
+    /// Runs a future to completion on the current runtime
+    pub fn block_on<F>(&self, future: F) -> F::Output
+    where
+        F: Future,
+    {
+        tokio::pin!(future);
 
         self.rt
             .block_on(self.local.run_until(crate::future::poll_fn(|cx| {
```

Comment:
Oh, self.local.run_until(poll_fun(... future.as_mut().poll ...))

That's a good synchronous stack to get one's async head around.